autogen/agentchat/groupchat.py

```diff
@@ -30,6 +30,7 @@ class GroupChat:
         - "manual": the next speaker is selected manually by user input.
         - "random": the next speaker is selected randomly.
         - "round_robin": the next speaker is selected in a round robin fashion, i.e., iterating in the same order as provided in `agents`.
+        - "graph": the next speaker is selected based on a graph. The select_speaker method is overridden by GraphGroupChat.
```

Comment:
I don't think "graph" should be a separate option here. I think it's better to think of the graph as a constraint, and in future we'll make it more general. So one can choose "auto", "manual", "random" or "round_robin" when such as constraint exists.

Reply:
Yes, that is possible too.

I infer that after contraining the candidate speaker(s), the methods are:
"auto" - use llm to select a speaker from the candidate(s).
"manual" - selected manually by the user input. What do we do if the user input is not as of the candidate(s)?
"random" - randomly select a speaker from the candidate(s).
"round_robin" - nodes are created sequentially (though I'm not sure whether I can get that sequence in `networkx`, pick the oldest node amongst the candidate(s), if I can't get the oldest node, get the first node by alphabetical order?

If the above is correct, do you still prefer that I inherit from `GroupChat` into `GraphGroupChat` and override `select_speaker` https://github.com/microsoft/autogen/blob/cb687dde8c0b3f8ee6d8c5634a53c5b2ccfed5b4/autogen/agentchat/contrib/graphgroupchat.py#L111 to implement the above `speaker_selection_method` (auto/manual/random/round_robin)?

Reply:
Resolving as with the summary:
"auto" - use llm to select a speaker from the candidate(s).
"manual" - selected manually by the user input. Only eligible transitions are allowed.
"random" - randomly select a speaker from the candidate(s).
"round_robin" - same order as agents, after filtering for eligible transitions.

GraphGroupChat is no longer a standalone class, all logic in `GroupChat`.